source

```
@@ -67350,6 +67379,58 @@ dictionary <dfn>ElementRegistrationOptions</dfn> {
    <li><p>Otherwise, return undefined.</p></li>
   </ol>
 
+  <p>When invoked, the <dfn><code
+  data-x="dom-CustomElementsRegistry-whenDefined">whenDefined(<var>name</var>)</code></dfn> method
+  must run these steps:</p>
+
+  <ol>
+   <li><p>If <var>name</var> is not a <span>valid custom element name</span>, then return a promise
+   rejected with a <code>SyntaxError</code> exception and abort these steps.</p></li>
+
+   <li><p>If this <code>CustomElementsRegistry</code> contains an entry with <span
+   data-x="concept-custom-element-definition-name">name</span> <var>name</var>, then return a
+   promise resolved with undefined and abort these steps.</p></li>
+
+   <li><p>Let <var>map</var> be this <code>CustomElementsRegistry</code>'s <span>when-defined
+   promise map</span>.</p></li>
+
+   <li><p>If <var>map</var> does not contain an entry with key <var>name</var>, create an entry in
+   <var>map</var> with key <var>name</var> and whose value is an empty list.</p></li>
+
+   <li><p>Let <var>promiseList</var> be the value of the entry in <var>map</var> with key
+   <var>name</var>.</p></li>
+
+   <li><p>Let <var>promise</var> be a new promise, and append <var>promise</var> to
+   <var>promiseList</var>.</p></li>
+
+   <li><p>Return <var>promise</var>.</p></li>
+  </ol>
```

Um, I'm not really sure. I guess it seems better to ensure a different promise is returned every time, instead of returning a new promise for the immediately-rejected/immediately-fulfilled cases, and a cached promise for the we-don't-know-yet cases?